argocd/schema_application_set.go

```diff
@@ -333,6 +339,12 @@ func applicationSetGitGeneratorSchemaV0() *schema.Schema {
 					MaxItems:    1,
 					Elem:        applicationSetTemplateResource(true),
 				},
+				"values": {
+					Type:        schema.TypeMap,
```

Comment:
```suggestion
					Type:        schema.TypeSet,
```

Using `TypeSet` is not strictly required but does prevent unnecessary updates if/when list order changes.